### ISSUE TITLE
Idea: Cave spider buffs

### ISSUE THREAD

**Issue by saltyseadoggo** [sf-inc/sihywtcamd#4]
I think that this mod could utilize cave spiders to enhance the danger of exploring caves, and/or make them a more prominent threat in their vanilla home, the mineshafts. In vanilla, cave spiders are restricted to only spawning from spawners, so the threat that they pose can be completely removed from a given area somewhat easily. This is compounded by the fact that the spawners often don't cover the area of the mineshaft very well due both to them not generating very frequently and needing the player to be within range of the.
I have some assorted ideas that you could potentially use or take inspiration to make them a bit more interesting. All of these would need to be subjected to playtesting, of course, for finer balance.

- **Cave spiders could spawn anywhere in caves.** They would have to have their spawn height limited to keep them underground. They could simply be a rare spawn anywhere underground, similarly to what this mod does with guardians in ocean biomes, but I think it would be more interesting if, as one goes lower in the world, less regular spiders and more cave spiders would spawn, making spelunking deep into the dark deadlier as one goes further down. The cave spider's tiny hitbox would also mean that they can spawn in 1x1x1 areas that other mobs cannot, making them potentially more insidious than regular mobs; players would find them occasionally in their branch mines and know that there's a tiny space somewhere they left unilluminated.

- **Cave spiders could use a longer or higher tier poison effect, or have a chance to.** Just making them all give Poison II would weaken the player faster, and that would be that. As an alternative, cave spiders could have a chance, based on local difficulty, to spawn with the ability to dish out longer-lasting or higher-tier poison effects. This alternative doesn't make much sense if they are not allowed to spawn in caves, however, since the spawners in mineshafts can be dealt with pretty quickly, so the local difficulty wouldn't have much a chance to kick in; if you keep them restricted to mineshafts, just giving them stronger/longer-lasting poison would make more sense.

**Cave spiders could deal more regular damage,** as they need to finish off a player due to their poison being unable to actually kill, and their current regular damage is pitiful, especially with armour, restricting their ability to do this.

- **Cave spider spawner corridors could be made more frequent in mineshafts.** This would make the mineshafts have more cave spiders without sacrificing the concept of needing to destroy their nests. Whether or not a mineshaft has plenty of spiders or not is based on worldgen RNG, so some are pitiful while others are deadly; as such, I'm not sure if this change is needed or not. 

- **Cave spiders spawned from spawners could be made more deadly** if they are allowed to spawn in caves, to make the structures more meaningful. They could have stronger or longer lasting poison, deal more regular damage, and/or even appear larger.

**Comment by charlyb01** [comment]
I love those ideas! Let me think about it more in details =)

**Comment by charlyb01** [comment]
Okay so I've made some testing and even though I loved the ideas of stronger spiders from spawners, I didn't implement it because I found them too deadly (even with a small buff). Cave spiders can spawn naturally in caves but will succeed to spawn more often if they're deeper. They can spawn in group from 1 to 3.  Finall, I didn't make a lot of changes that you suggested I'm sorry. However I added one thing (quite rare): cave spider jockey may spawn with a small chance (a bit more often then regular spider jockey as cave spiders spawn less often) but they can't spawn from spawners
